src/components/views/dialogs/InviteDialog.tsx

```diff
@@ -298,28 +299,45 @@ class DMRoomTile extends React.PureComponent {
     }
 }
 
-export default class InviteDialog extends React.PureComponent {
-    static propTypes = {
-        // Takes an array of user IDs/emails to invite.
-        onFinished: PropTypes.func.isRequired,
+interface IInviteDialogProps {
+    // Takes an array of user IDs/emails to invite.
+    onFinished: (toInvite?: string[]) => any;
 
-        // The kind of invite being performed. Assumed to be KIND_DM if
-        // not provided.
-        kind: PropTypes.string,
+    // The kind of invite being performed. Assumed to be KIND_DM if
+    // not provided.
+    kind: string,
 
-        // The room ID this dialog is for. Only required for KIND_INVITE.
-        roomId: PropTypes.string,
+    // The room ID this dialog is for. Only required for KIND_INVITE.
+    roomId: string,
 
-        // Initial value to populate the filter with
-        initialText: PropTypes.string,
-    };
+    // Initial value to populate the filter with
+    initialText: string,
+}
+
+interface IInviteDialogState {
+    targets: RoomMember[]; // array of Member objects (see interface above)
+    filterText: string;
+    recents: { user: DirectoryMember, userId: string }[];
+    numRecentsShown: number;
+    suggestions: { user: DirectoryMember, userId: string }[];
+    numSuggestionsShown: number;
+    serverResultsMixin: { user: DirectoryMember, userId: string }[];
+    threepidResultsMixin: ({ user: ThreepidMember, userId: string} | { user: DirectoryMember, userId: string})[];
```

Comment:
in theory these and all the other `DirectoryMember` definitions of the state can just use `Member` as the base class - there shouldn't be a need for them to use the specific types.

Reply:
ah yep, that makes it simpler